src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+							if (!superScrollView.ContentOffset.Equals(newContentOffset))
+							{
+								if (View?.Superview is UIStackView)
+									superScrollView.SetContentOffset(newContentOffset, UIView.AnimationsEnabled);
+								else
+									superScrollView.ContentOffset = newContentOffset;
+							}
+						}
+						superScrollView = superScrollView.Superview as UIScrollView;
+					}
+				});
+			}
+
+			if (IsKeyboardShowing)
+				RestorePosition();
+
+			IsKeyboardShowing = false;
+			View = null;
+			LastScrollView = null;
+			KeyboardFrame = CGRect.Empty;
+			StartingContentInsets = new UIEdgeInsets();
+			StartingScrollIndicatorInsets = new UIEdgeInsets();
+			StartingContentInsets = new UIEdgeInsets();
+		});
+	}
+
+	internal static void Destroy()
+	{
+		if (WillShowToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(WillShowToken);
+		if (DidHideToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(DidHideToken);
+		if (TextFieldToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextFieldToken);
+		if (TextViewToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextViewToken);
+	}
+
+	// Used to get the numeric values from the UserInfo dictionary's NSObject value to CGRect.
+	// Doing manually since CGRectFromString is not yet bound
+	static CGRect? DescriptionToCGRect(string description)
+	{
+		// example of passed in description: "NSRect: {{0, 586}, {430, 346}}"
+
+		if (description is null)
+			return null;
+
+		// remove letters in all languages, spaces, and curly brackets
+		var temp = Regex.Replace(description, @"[\p{L}\s:{}]", "");
+		var dimensions = temp.Split(',');
+
+		if (int.TryParse(dimensions[0], out var x) && int.TryParse(dimensions[1], out var y)
+			&& int.TryParse(dimensions[2], out var width) && int.TryParse(dimensions[3], out var height))
+		{
+			return new CGRect(x, y, width, height);
+		}
+
+		return null;
+	}
+
+	static async Task SetUpTextEdit()
+	{
+		if (View is null)
+			return;
+
+		CursorRect = null;
+
+		RootController = View.FindResponder<ContainerViewController>()?.View;
+
+		// the cursor needs a small amount of time to update the position
+		await Task.Delay(5);
+
+		UITextRange? selectedTextRange;
+		CGRect? localCursor = null;
+
+		if (View is UITextView tv)
+		{
+			selectedTextRange = tv.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+			{
+				localCursor = tv.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = tv.ConvertRectToView(local, null);
+			}
+		}
+		else if (View is UITextField tf)
+		{
+			selectedTextRange = tf.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+			{
+				localCursor = tf.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = tf.ConvertRectToView(local, null);
+			}
+		}
```

Comment:
The conde inside the ifs is very very very similar. This looks like we are looking at properties in the IUITextInput:

## IUITextInput 
* https://learn.microsoft.com/en-us/dotnet/api/uikit.iuitextinput.selectedtextrange?view=xamarin-ios-sdk-12#uikit-iuitextinput-selectedtextrange
* https://learn.microsoft.com/en-us/dotnet/api/uikit.iuitextinput.getcaretrectforposition?view=xamarin-ios-sdk-12#uikit-iuitextinput-getcaretrectforposition(uikit-uitextposition)

The rest of the code with the selectedRange is exactly the same, can you unify the ifs in a single one like this?
```suggestion
		if (View is IUITextInput tv)
		{
			selectedTextRange = tv.SelectedTextRange;
			if (selectedTextRange is UITextRange selectedRange)
			{
				localCursor = tv.GetCaretRectForPosition(selectedRange.Start);
				if (localCursor is CGRect local)
					CursorRect = tv.ConvertRectToView(local, null);
			}
		}
```
Less code, and shuoul

Reply:
Oh awesome! I was looking for something like this!